### PR TITLE
Documentation NimScript

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Improve Documentation of NimScript, update and expand.
- Mention the benefits aside of its limitations.
- Add examples.

Just Documentation only, if some section should be omitted let me know I will remove it.
The **Limitations** section at top of the NimScript page kinda looks discouraging for new people so I meant to add a section explaining that is not useless.

:slightly_smiling_face:

### CODE REVIEW

File: doc/nims.rst [hunk 1]
```diff
@@ -165,3 +196,129 @@ ends with ``.nims``:
   echo "hello world"
 
 Use ``#!/usr/bin/env -S nim --hints:off`` to disable hints.
+
+
+Benefits
+========
+
+**Cross-Platform**
```
Comment (narimiran) [inline doc/nims.rst]: I think these would work better as sub-headings, not as bold text:

```
Cross-Platform
--------------
```

File: doc/nims.rst [hunk 2]
```diff
@@ -47,18 +48,50 @@ NimScript is subject to some limitations caused by the implementation of the VM
...
 
 * `macros <macros.html>`_
 * `os <os.html>`_
 * `strutils <strutils.html>`_
 * `math <math.html>`_
 * `distros <distros.html>`_
-
-In addition to the standard Nim syntax (`system <system.html>`_
-module), NimScripts support the procs and templates defined in the
+* `sugar <sugar.html>`_
```
Comment (narimiran) [inline doc/nims.rst]: Nice job (testing and) adding all these!

File: doc/nims.rst [hunk 3]
```diff
@@ -47,18 +48,50 @@ NimScript is subject to some limitations caused by the implementation of the VM
...
+* `random <punycode.html>`_
+* `stats <stats.html>`_
+* `strformat <strformat.html>`_
+* `strmisc <strmisc.html>`_
+* `strscans <strscans.html>`_
+* `unicode <unicode.html>`_
+* `uri <uri.html>`_
+* `std/editdistance <editdistance.html>`_
+* `std/wordwrap <wordwrap.html>`_
+* std/sums
```
Comment (narimiran) [inline doc/nims.rst]: Do we need to create a documentation page for this one?

File: doc/nims.rst [hunk 4]
```diff
@@ -165,3 +196,129 @@ ends with ``.nims``:
...
 
 Use ``#!/usr/bin/env -S nim --hints:off`` to disable hints.
+
+
+Benefits
+========
+
+**Cross-Platform**
+
+Its a cross-platform scripting language that can run where Nim can run,
```
Comment (narimiran) [inline doc/nims.rst]: typo: Its --> It's (or It is)

File: doc/nims.rst (hunk 3 above)
Comment (juancarlospaco) [reply doc/nims.rst]: https://github.com/nim-lang/Nim/issues/11543#issue-458254530

Comment (narimiran) [reply doc/nims.rst]: [Fixed](https://github.com/nim-lang/Nim/commit/96bed814d2c630bee4eab8bd7a70b0975ada6581), you can add a link.

**Comment by juancarlospaco** [comment]
Let me know how it looks @narimiran :slightly_smiling_face:

File: doc/nims.rst [hunk 5]
```diff
@@ -165,3 +196,136 @@ ends with ``.nims``:
...
+Benefits
+========
+
+Cross-Platform
+--------------
+
+It is a cross-platform scripting language that can run where Nim can run,
+e.g. you can not run Batch or PowerShell on Linux or Mac,
+the Bash for Linux might not run on Mac,
+theres no Unittests tools for Batch, etc.
```
Comment (trustable-code) [inline doc/nims.rst]: I think this is spelled "unit tests".

File: doc/nims.rst [hunk 6]
```diff
@@ -165,3 +196,136 @@ ends with ``.nims``:
...
+
+NimScript can detect on which platform, operating system,
+architecture, and even which Linux Distribution is running on,
```
Comment (trustable-code) [inline doc/nims.rst]: I think this is spelled "Linux distribution".

File: doc/nims.rst [hunk 7]
```diff
@@ -165,3 +196,136 @@ ends with ``.nims``:
...
+that means there is nothing new to learn, no context switch for developers.
+
+
+Powerful Metaprogramming
+------------------------
+
+NimScript can use Nim's templates, macros, types, concepts, effect tracking system, and more,
+you can create modules that work on compiled Nim and also on interpreted NimScript.
+
+``func`` will still check for Side Effects, ``debugEcho`` also works as expected,
```
Comment (trustable-code) [inline doc/nims.rst]: I think this is spelled "side effects".

File: doc/nims.rst (hunk 5 above)
Comment (narimiran) [reply doc/nims.rst]: also "theres" --> "there's", but since it is plural --> "there are no unit test tools"

**Pull request merged**